src/App.tsx

```diff
@@ -1,204 +1,188 @@
 /* eslint-disable jsx-a11y/label-has-associated-control */
 /* eslint-disable jsx-a11y/control-has-associated-label */
-import React from 'react';
+import React, { useEffect, useState, useRef } from 'react';
 import { UserWarning } from './UserWarning';
 import { USER_ID } from './api/todos';
+import { Todo } from './types/Todo';
+import * as todoService from './api/todos';
+import { TodoList } from './components/todoList';
+import { TodoForm } from './components/todoForm';
+import { Filter } from './components/filter';
+import { Notification } from './components/notification';
+
+enum FilterType {
+  FILTERED_ACTIVE = 'Active',
+  FILTERED_COMPLETED = 'Completed',
+  CLEAR_COMPLETED = 'Clear completed',
+  UNFILTERED = 'All',
+}
 
 export const App: React.FC = () => {
+  const [isSubmitting, setIsSubmitting] = useState(false);
+
+  const [titleInput, setTitleInput] = useState('');
+
+  const [todos, setTodos] = useState<Todo[]>([]);
+
+  const [loading, setLoading] = useState(true);
+
+  const [errorMessage, setErrorMessage] = useState('');
+
+  const [hidden, setHidden] = useState(true);
+
+  const [filterBy, setFilterBy] = useState(FilterType.UNFILTERED);
+
+  const todoCount = todos.length;
+
+  function clear(todoId: number) {
+    todoService.deleteTodos(todoId).then(() => {
+      setTodos(currentTodos => currentTodos.filter(todo => todo.id !== todoId));
+    });
+  }
+
+  function getPreparedTodos(todoList: Todo[], filterType: string) {
+    const preparedTodos = [...todoList];
+
+    if (filterType === FilterType.UNFILTERED) {
```

Comment:
you compare `filterType` 3 times and then use else
That's the perfect case to use `switch` block

Also, this function gets all required data as arguments. That means you can move it out of the component (so JS will not create it every time the component is rerendering) or even into a separate file (utils.ts, etc.)